connectors/sources/graphql.py

```diff
@@ -0,0 +1,512 @@
+                "options": [
+                    {"label": "No Auth", "value": "none"},
+                    {"label": "Basic Auth", "value": "basic"},
+                    {"label": "Bearer Token", "value": "bearer"},
+                ],
+                "order": 3,
+                "type": "str",
+                "value": "none",
+            },
+            "username": {
+                "depends_on": [{"field": "authentication_method", "value": "basic"}],
+                "label": "Username",
+                "order": 4,
+                "type": "str",
+            },
+            "password": {
+                "depends_on": [{"field": "authentication_method", "value": "basic"}],
+                "label": "Password",
+                "order": 5,
+                "sensitive": True,
+                "type": "str",
+            },
+            "token": {
+                "depends_on": [{"field": "authentication_method", "value": "bearer"}],
+                "label": "Bearer Token",
+                "order": 6,
+                "sensitive": True,
+                "type": "str",
+            },
+            "graphql_query": {
+                "display": "textarea",
+                "label": "GraphQL Body",
+                "order": 7,
+                "type": "str",
+            },
+            "graphql_variables": {
+                "depends_on": [{"field": "http_method", "value": "post"}],
+                "display": "textarea",
+                "label": "Graphql Variables",
+                "order": 8,
+                "type": "str",
+                "required": False,
+            },
+            "graphql_object_list": {
+                "label": "GraphQL Objects List",
+                "order": 9,
+                "tooltip": "Specifies which GraphQL objects should be indexed as individual documents. This allows finer control over indexing, ensuring only relevant data sections from the GraphQL response are stored as separate documents. Use '.' to provide full path of the object from the root of the response. For example 'organization.users.nodes'",
+                "type": "list",
+            },
+            "headers": {
+                "label": "Headers",
+                "order": 10,
+                "type": "str",
+                "required": False,
+            },
+            "pagination_model": {
+                "display": "dropdown",
+                "label": "Pagination model",
+                "options": [
+                    {"label": "No pagination", "value": "no_pagination"},
+                    {"label": "Cursor-based pagination", "value": "cursor_pagination"},
+                ],
+                "order": 11,
+                "tooltip": "Cursor based pagination requires 'pageInfo' field along with argument 'after' for objects mentioned in 'GraphQL Objects List'. It also requires variable for 'after' argument.",
+                "type": "str",
+                "value": "none",
```

Comment:
```suggestion
                "value": "no_pagination",
```
if we want to default to `no pagination` we should set to `no_pagination`